src/Shell/Task/ModelTask.php

```diff
@@ -395,17 +427,69 @@ public function findBelongsToMany($model, array $associations)
                 $assocTable = substr($otherTableName, 0, $otherOffset);
             }
             if ($assocTable && in_array($assocTable, $tables)) {
-                $habtmName = $this->_camelize($assocTable);
-                $assoc = [
-                    'alias' => $habtmName,
-                    'foreignKey' => $foreignKey,
-                    'targetForeignKey' => $this->_modelKey($habtmName),
-                    'joinTable' => $otherTableName
-                ];
-                if ($assoc && $this->plugin) {
-                    $assoc['className'] = $this->plugin . '.' . $assoc['alias'];
+                $assocTableObject = TableRegistry::get($otherTableName);
+
+                $thisForeignKey = $primaryKey;
+                foreach ($thisForeignKey as $i => $fieldName) {
+                    if ($fieldName === 'id') {
+                        $thisForeignKey[$i] = Inflector::singularize($model->table()) . '_id';
+                    }
+                }
+                $otherForeignKeys = $assocTableObject->schema()->columns();
```

Comment:
E.g. this schema, to support composite foreign keys in bake could for instance look like this:
```sql
CREATE TABLE users_roles(
  site_id INT NOT NULL
, user_id INT NOT NULL 
, role_id INT NOT NULL
, PRIMARY KEY (site_id, user_id, role_id)
, FOREIGN KEY (site_id) REFERENCES sites (id) -- For belongsTo
, FOREIGN KEY (site_id, user_id) REFERENCES users (site_id, id) -- For belongsToMany
, FOREIGN KEY (site_id, role_id) REFERENCES roles (site_id, id) -- For belongsToMany
); 
```